### PR TITLE
14 retrieve the features support levels for given openshift version

### PR DESCRIPTION
**Pull request by clwheel**
Fix: #14

### CODE REVIEW

**Review by tonyskapunk — APPROVED** [review]
LGTM, thanks @clwheel for your contribution!

File: plugins/modules/features.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+QUERY_PARAMS_LIST = ["openshift_version", "cpu_architecture", "platform_type", "external_platform_name"]
+
+def run_module():
+    module_args = dict(
+        openshift_version = dict(type="str", required=True),
+        cpu_architecture = dict(type="str", required=False),
+        platform_type = dict(type="str", required=False),
```
Comment (betoredhat) [inline plugins/modules/features.py:97]: I think a couple of this does not match with the documentation

File: plugins/modules/features.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    description: The CPU architecture of the OpenShift version (e.g. x86_64, aarch64, arm64, ppc64le, s390x, multi)
+    default: None
+    required: False
+    type: string
+
+  platform_type:
+    description: The provider platform type (e.g. baremetal, none, nutanix, vsphere, external)
+    default: None
+    required: True
+    type: string
```
Comment (tonyskapunk) [inline plugins/modules/features.py:39]: Please add choices:

```suggestion
    choices: [ baremetal, none, nutanix, vsphere, external ]
    type: string
```

File: plugins/modules/features.py (hunk 1 above)
Comment (tonyskapunk) [inline plugins/modules/features.py:97]: Please add its choices:
```suggestion
        platform_type = dict(type="str", required=False, choices=["baremetal", "none", "nutanix", "vsphere", "external"]), 
```

File: plugins/modules/features.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+from ansible.module_utils.basic import AnsibleModule
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+QUERY_PARAMS_LIST = ["openshift_version", "cpu_architecture", "platform_type", "external_platform_name"]
+
+def run_module():
+    module_args = dict(
+        openshift_version = dict(type="str", required=True),
+        cpu_architecture = dict(type="str", required=False),
```
Comment (tonyskapunk) [inline plugins/modules/features.py:96]: Please add choices and its default:
```suggestion
        cpu_architecture = dict(type="str", required=False, default="x86_64", choices=["x86_64", "aarch64", "arm64", "ppc64le", "s390x", "multi"]),
```

**Review by tonyskapunk — CHANGES_REQUESTED** [review]

**Review by tonyskapunk — APPROVED** [review]
LGTM, thanks @clwheel !!

File: plugins/modules/features.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    cpu_architecture: x86_64
+    platform_type: baremetal
+"""
+
+RETURN = r"""
+features:
+  description: A list of supported OpenShift features
+  type: dict
+  returned: always
+  sample: { "CLUSTER_MANAGED_NETWORKING": "supported",
```
Comment (vjayaramrh) [inline plugins/modules/features.py:59]: I saw the below as output when I curled, looks like the response is a dict mapped to the `features` key, should the same be updated to reflect that? CC: @clwheel @tonyskapunk 
```
[openshift@openshift assistedinstaller]$ curl --location https://api.openshift.com/api/assisted-install/v2/support-levels/features?openshift_version=4.16 --header "Authorization: Bearer ${token}" 
{"features":{"BAREMETAL_PLATFORM":"supported","CLUSTER_MANAGED_NETWORKING":"supported","CNV":"supported","CUSTOM_MANIFEST":"supported","DUAL_STACK":"supported","DUAL_STACK_VIPS":"supported","EXTERNAL_PLATFORM":"supported","EXTERNAL_PLATFORM_OCI":"supported","FULL_ISO":"supported","LSO":"supported","LVM":"supported","MCE":"supported","MINIMAL_ISO":"supported","NONE_PLATFORM":"supported","NUTANIX_INTEGRATION":"supported","ODF":"supported","OVN_NETWORK_TYPE":"supported","SDN_NETWORK_TYPE":"unavailable","SINGLE_NODE_EXPANSION":"supported","SKIP_MCO_REBOOT":"supported","SNO":"supported","USER_MANAGED_NETWORKING":"supported","VIP_AUTO_ALLOC":"unavailable","VSPHERE_INTEGRATION":"supported"}}

```

Comment (clwheel) [reply plugins/modules/features.py:59]: @vjayaramrh If you look at [line 55](https://github.com/vjayaramrh/assistedinstaller/pull/24/files/8240ef2b0f708f38dca7aa5b85d3629f7819cc8c#diff-bc610870e170ba96b364b2f7c7fc668fb3e1e43d96b4c87eb608fd04f1f3a628R55) it shows that `features` is the key of the dictionary that is returned.

Comment (vjayaramrh) [reply plugins/modules/features.py:59]: Thanks @clwheel

**Pull request merged**